File: src/chrome/android/java/src/org/chromium/chrome/browser/customtabs/ContentPrivacyPreferencesBridge.java
// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.customtabs;

import org.jni_zero.NativeMethods;

/**
 * Bridge class to access content privacy preferences from C++ to Java.
 * This allows Java code to check the SAML-controlled content privacy setting.
 */
public class ContentPrivacyPreferencesBridge {
    
    /**
     * Get the current content privacy enabled state from native preferences.
     * @return true if content privacy is enabled (screenshots should be blocked), false otherwise
     */
    public static boolean isContentPrivacyEnabled() {
        return ContentPrivacyPreferencesBridgeJni.get().isContentPrivacyEnabled();
    }
    
    @NativeMethods
    interface Natives {
        /**
         * Returns whether content privacy is enabled from SAML preferences.
         * @return true if content privacy is enabled, false otherwise (default: false)
         */
        boolean isContentPrivacyEnabled();
    }
} 

File: src/chrome/browser/android/content_privacy_preferences_bridge.cc
// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/android/jni_android.h"
#include "chrome/browser/profiles/profile.h"
#include "chrome/browser/profiles/profile_manager.h"
#include "content/public/browser/content_privacy_prefs.h"
#include "components/prefs/pref_service.h"

// Must come after all headers that specialize FromJniType() / ToJniType().
#include "chrome/android/chrome_jni_headers/ContentPrivacyPreferencesBridge_jni.h"

static jboolean JNI_ContentPrivacyPreferencesBridge_IsContentPrivacyEnabled(JNIEnv* env) {
  Profile* profile = ProfileManager::GetActiveUserProfile();
  if (!profile) {
    return false;  // Default to disabled if no profile
  }
  
  PrefService* prefs = profile->GetPrefs();
  if (!prefs) {
    return false;  // Default to disabled if no prefs
  }
  
  // Get the content privacy enabled preference (defaults to false)
  return prefs->GetBoolean(content_privacy::prefs::kContentPrivacyEnabled);
} 